test/jasmine/tests/image_test.js

```diff
@@ -0,0 +1,381 @@
+            type: 'image',
+            z: [[[2, 1], [4, 3]]],
+            contours: {coloring: 'lines'}
+        }])
+        .then(function() {
+            expect(getIndices()).toEqual([0, 1]);
+            return Plotly.restyle(gd, 'visible', false, [0]);
+        })
+        .then(function() {
+            expect(getIndices()).toEqual([1]);
+            return Plotly.restyle(gd, 'visible', true, [0]);
+        })
+        .then(function() {
+            expect(getIndices()).toEqual([0, 1]);
+        })
+        .catch(failTest)
+        .then(done);
+    });
+});
+
+describe('image hover:', function() {
+    'use strict';
+
+    var gd;
+
+    describe('for `image_cat`', function() {
+        beforeAll(function(done) {
+            gd = createGraphDiv();
+
+            var mock = require('@mocks/image_cat.json');
+            var mockCopy = Lib.extendDeep({}, mock);
+
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+        });
+
+        afterAll(destroyGraphDiv);
+
+        function _hover(gd, xval, yval) {
+            var fullLayout = gd._fullLayout;
+            var calcData = gd.calcdata;
+            var hoverData = [];
+
+            for(var i = 0; i < calcData.length; i++) {
+                var pointData = {
+                    index: false,
+                    distance: 20,
+                    cd: calcData[i],
+                    trace: calcData[i][0].trace,
+                    xa: fullLayout.xaxis,
+                    ya: fullLayout.yaxis
+                };
+
+                var hoverPoint = Image.hoverPoints(pointData, xval, yval);
+                if(hoverPoint) hoverData.push(hoverPoint[0]);
+            }
+
+            return hoverData;
+        }
+
+        it('should find closest point (case 1) and should', function() {
+            var pt = _hover(gd, 0, 0)[0];
+            expect(pt.index).toEqual([0, 0], 'have correct index');
+        });
+
+        it('should find closest point (case 2) and should', function() {
+            var pt = _hover(gd, 50, 0)[0];
+            expect(pt.index).toEqual([0, 50], 'have correct index');
+        });
+    });
+
+    describe('for `image_adventurer`', function() {
+        var mock = require('@mocks/image_adventurer.json');
+        beforeAll(function() {
+            gd = createGraphDiv();
+        });
+
+        afterAll(destroyGraphDiv);
+
+        function _hover(x, y) {
+            var evt = { xpx: x, ypx: y };
+            return Fx.hover('graph', evt, 'xy');
+        }
+
+        it('should display RGB channel values', function(done) {
+            var mockCopy = Lib.extendDeep({}, mock);
+            mockCopy.data[0].colormodel = 'rgb';
+            Plotly.newPlot(gd, mockCopy)
+            .then(function() {_hover(205, 125);})
+            .then(function() {
+                assertHoverLabelContent({
+                    nums: 'x: 25.5\ny: 14.5\nz: [54, 136, 153]\n<tspan style="text-transform:uppercase">rgb</tspan>: [54, 136, 153]',
```

Comment:
So you're showing the color by default. I thought we decided on just `x+y+z+text` in last week's meeting. Please correct me if I'm wrong.

Reply:
Yes, the code currently shows the color by default. You're right that this isn't what we discussed in the meeting. I'm OK with removing `color` but I also think it could be a default. It's your call :)

cc @nicolaskruchten @etpinard

Reply:
Having `x+y+z+color+all` as default is pretty nice when one sets `zmin` and `zmax`, but otherwise the `color` flag is essentially that same as `z` flag.

So, I'm thinking we should either:

1) set the `hoverinfo` default to `x+y+z+color+all` when `zmin` and/or `zmax` is set, fallback to `x+y+z+all` otherwise
2) consider that `zmin` and `zmax` won't be set very often by users and then `x+y+z+all` is a fine "hard" default for `hoverinfo`.

---

I'd personally vote for 2)

Reply:
Reading your comment, I realized that the trace lacked `hovertext` and `text` attributes. I added both in fd51e206f7bc3cdc6b4c6b13559c7a6535c771a3

Reply:
Ok, so I went with your suggestion number 2.

The change is introduced in e4cac27013813e94a4439809d2de1c53e90c0f95 along with a test!

Reply:
Awesome :tada: